docs/csharp/roslyn/work-with-syntax.md

```diff
@@ -0,0 +1,102 @@
+
+Unlike syntax nodes and tokens, syntax trivia do not have parents. Yet, because they are part of the tree and each is associated with a single token, you may access the token it is associated with using the <xref:Microsoft.CodeAnalysis.SyntaxTrivia.Token> property.
+
+Like syntax tokens, trivia are value types. The single <xref:Microsoft.CodeAnalysis.SyntaxTrivia> type is used to describe all kinds of trivia.
+
+## Spans
+
+Each node, token, or trivia knows its position within the source text and the number of characters it consists of. A text position is represented as a 32-bit integer, which is a zero-based Unicode character index. A <xref:Microsoft.CodeAnalysis.Text.TextSpan> object is the beginning position and a count of characters, both represented as integers. If <xref:Microsoft.CodeAnalysis.Text.TextSpan> has a zero length, it refers to a location between two characters.
+
+Each node has two <xref:Microsoft.CodeAnalysis.Text.TextSpan> properties: <xref:Microsoft.CodeAnalysis.Text.TextSpan.Span> and <xref:Microsoft.CodeAnalysis.Text.TextSpan.FullSpan>. 
+
+The <xref:Microsoft.CodeAnalysis.Text.TextSpan.Span> property is the text span from the start of the first token in the node’s sub-tree to the end of the last token. This span does not include any leading or trailing trivia.
+
+The <xref:Microsoft.CodeAnalysis.Text.TextSpan.FullSpan> property is the text span that includes the node’s normal span, plus the span of any leading or trailing trivia.
+
+For example: 
+
+``` csharp
+      if (x > 3)
+      {
+||        // this is bad
+          |throw new Exception("Not right.");|  // better exception?||
+      }
+```
+
+The statement node inside the block has a span indicated by the single vertical bars (|). It includes the characters `throw new Exception(“Not right.”);`. The full span is indicated by the double vertical bars (||). It includes the same characters as the span and the characters associated with the leading and trailing trivia.
+
+## Kinds
+
+Each node, token, or trivia has a <xref:Microsoft.CodeAnalysis.SyntaxNode.RawKind> property, of type <xref:System.Int32?displayProperty=fullName>, that identifies the exact syntax element represented. This value can be cast to a language-specific enumeration; each language, C# or VB, has a single <xref:Microsoft.CodeAnalysis.CSharp.SyntaxKind> enumeration that lists all the possible nodes, tokens, and trivia elements in the grammar. This conversion can be done automatically by accessing the <xref:Microsoft.CodeAnalysis.CSharp.CSharpExtensions.Kind?displayProperty=nameWithType> or <xref:Microsoft.CodeAnalysis.VisualBasic.VisualBasicExtensions.Kind?displayProperty=nameWithType> extension methods.
```

Comment:
```
each language, C# or VB, has a single <xref:Microsoft.CodeAnalysis.CSharp.SyntaxKind> enumeration
```

I find it weird that the text talks about both languages, but the link is then C#-specific. What about the following?

```
each language, C# or VB, has a single `SyntaxKind` enumeration 
(<xref:Microsoft.CodeAnalysis.CSharp.SyntaxKind?displayProperty=fullName> and 
<xref:Microsoft.CodeAnalysis.VisualBasic.SyntaxKind?displayProperty=fullName>, respectively)
```